modules/tutorials/partials/migration-tutorial/extract-sql-data.adoc

```diff
@@ -0,0 +1,83 @@
+[#extract-sql-data]
+== Extract your Course data from MySQL
+
+The first stage of your migration is to extract the data a file format that the `cbimport` utility can work with.
+`cbimport` can work with comma-separated value files or JSON-formatted files.
+Because we already know that we will be embedding our `enrollment` records into the record for each student,
+makes sense to use the more versatile JSON structure. 
+
+Fortunately, MySQL has a number of SQL functions that make working with JSON data fairly straightforward,
+so we'll start by migrating the `course` table into a JSON file:
+
+[source, mysql]
+.Extract the `course` table
+----
+SELECT JSON_OBJECT(
+       'course-id', course.`course-id`,
+       'course-name', course.`course-name`,
+       'faculty', course.faculty,
+       'credit-points', course.`credit-points`
+       ) FROM course
+INTO OUTFILE '/var/lib/mysql-files/courses.json'
+----
+
+Using the `JSON_OBJECT` function, the command will `SELECT` every record in the table and output it to a file.
+Each line of the file will correspond to a single record:
+
+[source,jsonlines]
+----
+{"faculty": "Art", "course-id": 1, "course-name": "Art History", "credit-points": 50}
+{"faculty": "Art", "course-id": 2, "course-name": "Fine Art", "credit-points": 30}
+{"faculty": "Design", "course-id": 3, "course-name": "Graphic Design", "credit-points": 70}
+{"faculty": "English", "course-id": 4, "course-name": "Creative Writing", "credit-points": 70}
+----
+
+NOTE: Strictly speaking, the JSON output is not a well-formed JSON document because it isn't structured as an array.
+Nevertheless, `cbimport` will read each line as a separate record.
+
+== Extract your Student data from MySQL
+
+This case is slightly different because we want to include the enrollment details with each student record 
+
+
+We can handle this JSON structure by using a more involved SELECT: 
+as well as extracting the student records, we can simultaneously pull in the enrollments for each student:
```

Comment:
We can handle this JSON structure by using a more involved `SELECT`.

As well as extracting the student records, we can simultaneously pull in the enrollments for each student: